De ant/aws migration (#149)
OpenDataCubePipelines/ard-pipeline: README.md

The [bitshuffle filter](https://github.com/kiyo-masui/bitshuffle) can be installed
from source, or conda via the supplied [conda recipe](https://github.com/kiyo-masui/bitshuffle/tree/master/conda-recipe).
It utilises a bitshuffling filter on top of either a lz4 or lzf compression filter.

# Digital Earth Antarctica's ARD Pipeline on AWS

> This section explains how to setup and run the ARD pipeline on an AWS EC2 instance running Ubuntu 22 and higher distribution. This approach is specifically designed for Antarctica (As part od the ARD Pipeline for GA's DE Antarctica) but it could be easily expanded to any other region.

Reviewer: ```suggestion
This section explains how to setup and run the ARD pipeline on an AWS EC2 instance running Ubuntu 22 and higher distributions. This approach is specifically developed for GA's _DE Antarctica_ Optical ARD Pipeline, but can be easily expanded to any other region.
```

## Setup

There are a few prerequisites before you can run a ARD Pipeline job which are listed below:

* ARD-Pipeline Project setup
* MODTRAN setup
* ARD-Pipeline downloader setup

Follow the steps in each section to setup your environment for running the pipeline.

**Note:** This guideline assumes that you are setting up and running the ARD Pipeline on an AWS EC2 machine running Ubuntu 22 or higher distributions.

### ARD-Pipeline project setup

* Clone the ARD-pipeline repository from here [ARD Pipeline](https://github.com/OpenDataCubePipelines/ard-pipeline.git) to your home directory and check out the `dea-ant/aws-migration` branch.

You can find your home directory by running `echo $HOME`.

> * Create and activate the conda environment dor the pipeline project by running

Reviewer: ```suggestion
* Create and activate the project's conda environment by running
```

```bash
conda create -f deployment/environment.yaml
```
and then by
```bash
conda activate ard-pipeline
```

You need a `conda` distribution to be able to create and activate conda environment. `Miniconda` distribution is the recommended one and it will be used in this guideline.

* Run the following
```bash
pip install --no-binary :all: \
  "git+https://github.com/ubarsc/rios@rios-${rios_version:-1.4.10}#egg=rios" \
  "git+https://github.com/ubarsc/python-fmask@pythonfmask-${fmask_version:-0.5.7}#egg=python-fmask" \
  awscli boto boto3
```

> This command will install additional requirements for the pipeline.

Reviewer: ```suggestion
* Run the following to install additional pipeline dependencies:
```bash
pip install --no-binary :all: \
  "git+https://github.com/ubarsc/rios@rios-${rios_version:-1.4.10}#egg=rios" \
  "git+https://github.com/ubarsc/python-fmask@pythonfmask-${fmask_version:-0.5.7}#egg=python-fmask" \
  awscli boto boto3
```
```

* Install WAGL using the command:
```bash
pip install --no-build-isolation --editable .
```

> * *OPTIONAL*: I you want to contribute to the code base, you will need to install the `pre-commit` library via `pip install pre-commit`

Reviewer: ```suggestion
*OPTIONAL*: If you want to contribute to the code base, please install the `pre-commit` library with `pip install pre-commit`.
```

### MODTRAN setup

> The pipeline requires [MODTRAN Software](http://modtran.spectral.com/modtran_index) for processing atmospheric information. To sut up a working copy of MODTRAN follow the steps below:

Reviewer: ```suggestion
This section assumes users have access to various GA's S3 buckets on AWS.

The pipeline requires [MODTRAN Software](http://modtran.spectral.com/modtran_index) for processing atmospheric information. To set up a working copy of MODTRAN follow the steps below:
```

> * Copy MODTRAN 6 to you home directory from [AWS S3](s3://imam-dev-bucket/LS9-isolated-pipeline/volumes/ancillary/MODTRAN6.0.2.3G/) by running the command below (assuming that you are in you home directory):

Reviewer: ```suggestion
Copy MODTRAN 6 to your home directory from [AWS S3](s3://imam-dev-bucket/LS9-isolated-pipeline/volumes/ancillary/MODTRAN6.0.2.3G/) by running the command below (assuming that you are in your home directory & have AWS CLI installed):
```

```bash
aws s3 cp s3://imam-dev-bucket/LS9-isolated-pipeline/volumes/ancillary/MODTRAN6.0.2.3G/ MODTRAN6/ --recursive
```
The total downloaded size of the data is about 30GB, so depending on your network connection speed, it might take some time to download the data.

* Add MODTRAN to the path and give it the right permissions.

```bash
export PATH="$HOME/MODTRAN6/bin/linux:$PATH"
sudo chmod -R a+x $HOME/MODTRAN6
```

* Activate MODTRAN using your license key

```bash
$HOME/MODTRAN6/bin/linux/mod6c_cons -activate_license <MODTRAN_PRODUCT_KEY>
```

You can also check your MODTRAN Software version by running

```bash
$HOME/MODTRAN6/bin/linux/mod6c_cons -version
```

which should show an output similar to this (depending n your software version): `MODTRAN(R) 6.0.2.3G`

### Checking the environment

> At this stage you can check if the environment is set up successfully buy running `./deployment/check-environment.sh` while you are in your ard-pipeline conda environment.

Reviewer: ```suggestion
At this stage you can check if the environment is configured correctly by running `./deployment/check-environment.sh` while you are in your ard-pipeline conda environment.
```

You should be able to see results similar to this:

```bash
Checking environment...
Trying rasterio... ✅ 1.3.9
Trying luigi... ✅ 3.5.0
Trying wagl... ✅ 6.1.2.dev244+gf001cc1f.d20260529
Attempting load of fortran-based modules... ✅
Attempting hdf5 blosc compression...✅
Checking modtran is available...✅
```

### ARD Pipeline Ancillary Downloader setup

> To run the ARD pipeline on AWS, you will need to use the ancillary data downloader package, designed to access and download the required ancillary files from various sources. For Digital Earth Antarctica specific pipeline, this data includes BRD, ERA45 and MERRA2 files required for running the pipeline over Antarctica. You need to clone this repo from here [ARD Pipeline Downloader](https://github.com/arcisad/ard-pipeline-downloader.git) to your **home** directory and create the conda environment for it using the `deployment/environment.yaml` file provided in its root directory. You don;t need to activate its conda environment for running the ARD pipeline.

Reviewer: ```suggestion

To run the ARD pipeline on AWS, you will need to use the `ard-pipeline-downloader` utility, designed to access and download the required ancillary files from various sources. For the Digital Earth Antarctica pipeline, this includes BRDF, ERA5 and MERRA2 data files. You need to clone this repo [ARD Pipeline Downloader](https://github.com/arcisad/ard-pipeline-downloader.git) to your **home** directory and create the conda environment for it using the `deployment/environment.yaml` file provided in its root directory. You don't need to activate its conda environment for running the ARD pipeline.
```

### ARD-Pipeline project setup

* Clone the ARD-pipeline repository from here [ARD Pipeline](https://github.com/OpenDataCubePipelines/ard-pipeline.git) to your home directory and check out the `dea-ant/aws-migration` branch.

You can find your home directory by running `echo $HOME`.

* Create and activate the project's conda environment by running

```bash
conda create -f deployment/environment.yaml
```
and then by
```bash
conda activate ard-pipeline
```

You need a `conda` distribution to be able to create and activate conda environment. `Miniconda` distribution is the recommended one and it will be used in this guideline.

* Run the following to install additional pipeline dependencies:
```bash
pip install --no-binary :all: \
  "git+https://github.com/ubarsc/rios@rios-${rios_version:-1.4.10}#egg=rios" \
  "git+https://github.com/ubarsc/python-fmask@pythonfmask-${fmask_version:-0.5.7}#egg=python-fmask" \
  awscli boto boto3
```

* Install WAGL using the command:
```bash
pip install --no-build-isolation --editable .
```

* *OPTIONAL*: If you want to contribute to the code base, please install the `pre-commit` library with `pip install pre-commit`

### MODTRAN setup

> This section assumes users have access to various GA's S3 buckets on AWS.

Reviewer: ```suggestion
This section assumes users have access to various GA S3 buckets on AWS.
```
I missed a typo in my own edit...
